client/apps/edit/components/admin/test/article.coffee

```diff
@@ -0,0 +1,176 @@
+benv = require 'benv'
+sinon = require 'sinon'
+{ resolve } = require 'path'
+React = require 'react'
+ReactDOM = require 'react-dom'
+ReactTestUtils = require 'react-addons-test-utils'
+fixtures = require '../../../../../../test/helpers/fixtures.coffee'
+Article = require '../../../../../models/article.coffee'
+Backbone = require 'backbone'
+moment = require 'moment'
+
+r =
+  find: ReactTestUtils.scryRenderedDOMComponentsWithClass
+  findTag: ReactTestUtils.scryRenderedDOMComponentsWithTag
+  simulate: ReactTestUtils.Simulate
+
+describe 'AdminArticle', ->
+
+  beforeEach (done) ->
+    benv.setup =>
+      benv.expose
+        $: benv.require 'jquery'
+        Bloodhound: (@Bloodhound = sinon.stub()).returns(
+          initialize: ->
+          ttAdapter: ->
+        )
+      $.fn.typeahead = sinon.stub()
+      AdminArticle = benv.require resolve __dirname, '../article/index.coffee'
+      AdminArticle.__set__ 'sd', {
+        API_URL: 'http://localhost:3005/api'
+        CURRENT_CHANNEL: id: '123'
+        USER: access_token: ''
+      }
+      AdminArticle.__set__ 'AutocompleteList', @Autocomplete = benv.require resolve __dirname, '../../../../../components/autocomplete_list/index.coffee'
+      @Autocomplete.__set__ 'request', get: sinon.stub().returns
+        set: sinon.stub().returns
+          end: sinon.stub().yields(null, body: { id: '123', name: 'Molly Gottschalk'})
+      @article = new Article
+      @article.attributes = fixtures().articles
+      @article.set('author', {name: 'Artsy Editorial', id: '123'})
+      @article.set('contributing_authors', [{name: 'Molly Gottschalk', id: '123'}])
+      @article.set('featured', true)
+      props = {
+        article: @article
+        onChange: sinon.stub()
+        }
+      @component = ReactDOM.render React.createElement(AdminArticle, props), (@$el = $ "<div></div>")[0], =>
+        setTimeout =>
+          done()
+
+  afterEach ->
+    benv.teardown()
+
+
+  describe 'Display', ->
+
+    it 'Renders the primary author field', ->
+      $(ReactDOM.findDOMNode(@component)).find('input').first().val().should.eql 'Artsy Editorial'
+      $(ReactDOM.findDOMNode(@component)).find('input').first().attr('placeholder').should.eql 'Change Primary Author name'
+
+    it 'Renders the contributing author field', ->
+      $(ReactDOM.findDOMNode(@component)).find('.autocomplete-input').first().attr('placeholder').should.eql 'Search by user name or email...'
+      $(ReactDOM.findDOMNode(@component)).text().should.containEql 'Contributing Author'
+      $(ReactDOM.findDOMNode(@component)).find('.autocomplete-select-selected').text().should.containEql 'Molly Gottschalk'
```

Comment:
This looks good for render tests, but we should also test callback code such as `selected` and `removed` since that's the heftiest work outside of the main Autocomplete functions. I think it could be as simple as finding the Autocomplete component and calling the functions directly as opposed to simulating a click. Does that make sense? I'm fine with adding those tests afterwards since this PR is already pretty big!